semver.py

```diff
@@ -174,6 +174,39 @@ def __iter__(self):
         for v in self._astuple():
             yield v
 
+    def __getitem__(self, index):
+        """Implement getitem. If the part requested is undefined, or a part of the
+           range requested is undefined, it will throw an index error.
+           Negative indices are not supported
+
+           :param Union[int, slice] index: a positive integer indicating the
+                  offset or a slice
+           :raises: IndexError, if index is beyond the range or a part is None
+           """
+        undefined_error = IndexError("Version part undefined")
+        negative_error = IndexError("Version index cannot be negative")
+        if isinstance(index, slice):
+            if (False if index.start is None else index.start < 0) or \
+                    (False if index.stop is None else index.stop < 0):
+                raise negative_error
+            else:
+                slice_is_full = True
+                part = self._astuple()[index]
+                for i in part:
```

Comment:
What is `i` here? Pylint would be complaining (if you ran it). Can you use a self-explanatory name?

Reply:
One idea could be to replace `i` with `item`.

Reply:
I guess I could use `subpart` instead.

Reply:
See also the more recent comments below. :wink: